chips/rp2040/src/pio.rs

```diff
@@ -515,54 +570,15 @@ pub enum SMNumber {
     SM3 = 3,
 }
 
+const SM_NUMBERS: [SMNumber; 4] = [SMNumber::SM0, SMNumber::SM1, SMNumber::SM2, SMNumber::SM3];
```

Comment:
What is the reason for this constant?

Reply:
@j-kk can you document this?

I _think_ it's just the list of state machine names on the chip

Reply:
Yes, it is just used for convinience in pio constructors

Reply:
We generally like to keep this documentation in the source code. Unfortunately, GitHub issue / PR comments are pretty hard to keep track of alongside the code. Can you please transfer these suggestions into rustdoc-comments (`///`) in the source? Thanks!